src/subscription.ts

```diff
@@ -892,7 +892,11 @@ export class Subscription extends EventEmitter {
     if (typeof snapshot === 'string') {
       reqOpts.snapshot = Snapshot.formatName_(this.pubsub.projectId, snapshot);
     } else if (Object.prototype.toString.call(snapshot) === '[object Date]') {
-      reqOpts.time = snapshot as google.protobuf.ITimestamp;
+      const dateMillis = (snapshot as Date).getTime();
+      reqOpts.time = {
+        seconds: Math.floor(dateMillis / 1000),
```

Comment:
Part of me thinks [`process.hrtime()`](https://nodejs.org/api/process.html#process_process_hrtime_time) would read better, at the same time it makes things harder to webpack ... so I lean towards your approach.

Reply:
Hmm, I didn't know about hrtime(). That's pretty nice. I think I'll go ahead and keep it as it is though, for the reason you gave. I don't think we support using the libraries from a browser anyway, right? But it doesn't seem good to introduce future problems just in case.